doc/classes/@GlobalScope.xml

```diff
@@ -2728,14 +2728,61 @@
 		<constant name="PROPERTY_HINT_OBJECT_ID" value="22" enum="PropertyHint">
 		</constant>
 		<constant name="PROPERTY_HINT_TYPE_STRING" value="23" enum="PropertyHint">
-			Hint that a property represents a particular type. If a property is [constant TYPE_STRING], allows to set a type from the create dialog. If you need to create an [Array] to contain elements of a specific type, the [code]hint_string[/code] must encode nested types using [code]":"[/code] and [code]"/"[/code] for specifying [Resource] types. For instance:
-			[codeblock]
-			hint_string = "%s:" % [TYPE_INT] # Array of integers.
-			hint_string = "%s:%s:" % [TYPE_ARRAY, TYPE_REAL] # Two-dimensional array of floats.
-			hint_string = "%s/%s:Resource" % [TYPE_OBJECT, TYPE_OBJECT] # Array of resources.
-			hint_string = "%s:%s/%s:Resource" % [TYPE_ARRAY, TYPE_OBJECT, TYPE_OBJECT] # Two-dimensional array of resources.
-			[/codeblock]
-			[b]Note:[/b] The final colon is required for properly detecting built-in types.
+			If a property is [String], hints that the property represents a particular type (class). This allows to select a type from the create dialog. The property will store the selected type as a string.
+			If a property is [Array], hints the editor how to show elements. The [code]hint_string[/code] must encode nested types using [code]":"[/code] and [code]"/"[/code].
+			[codeblocks]
+			[gdscript]
+			# Array of elem_type.
+			hint_string = "%d:" % [elem_type]
+			hint_string = "%d/%d:%s" % [elem_type, elem_hint, elem_hint_string]
+			# Two-dimensional array of elem_type.
```

Comment:
Actually, it's an array of arrays, not a two-dimensional array. Should we be worried about this?

Reply:
Added clarification.